multicluster/controllers/multicluster/commonarea/remote_common_area.go

```diff
@@ -409,6 +408,17 @@ func (r *remoteCommonArea) StopWatching() {
 	}
 	r.managerStopFunc()
 	r.managerStopFunc = nil
+
+	// Reset ClusterManager so this common area can be started again when it's reconnected.
+	mgr, err := ctrl.NewManager(r.config, ctrl.Options{
```

Comment:
Do we need to create a new one, or we can also reset some state in the existing Manager?

Reply:
I tried to figure out a way to reuse existing Mananger, but looks like it doesn't work. It relies on the framework of controller-runtime, if I set  one option `GracefulShutdownTimeout` to zero as a work around to skip the error `can't accept new runnable as stop procedure is already engaged`, it will move on and goes to here https://github.com/kubernetes-sigs/controller-runtime/blob/188b48630db275131d0fef1818933d8c2da96753/pkg/manager/internal.go#L489-L493 when we are trying to start the manager again, but it will raise `close of a closed channel` because it's closed when the manager received stop cancel context at first time.

Reply:
Ok. Creating a new one is fine. Just want to understand.